contracts/modules/andromeda-curve/src/contract.rs

```diff
@@ -0,0 +1,212 @@
+};
+use andromeda_modules::curve::{
+    CurveConfig, CurveId, CurveRestriction, ExecuteMsg, GetCurveConfigResponse,
+    GetPlotYFromXResponse, GetRestrictionResponse, InstantiateMsg, QueryMsg,
+};
+use andromeda_std::{
+    ado_base::{InstantiateMsg as BaseInstantiateMsg, MigrateMsg},
+    ado_contract::ADOContract,
+    common::{actions::call_action, context::ExecuteContext, encode_binary},
+    error::ContractError,
+};
+
+use cosmwasm_std::{
+    ensure, entry_point, Addr, Binary, Deps, DepsMut, Env, MessageInfo, Response, Storage,
+};
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:andromeda-curve";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    let resp = ADOContract::default().instantiate(
+        deps.storage,
+        env,
+        deps.api,
+        &deps.querier,
+        info,
+        BaseInstantiateMsg {
+            ado_type: CONTRACT_NAME.to_string(),
+            ado_version: CONTRACT_VERSION.to_string(),
+            kernel_address: msg.kernel_address,
+            owner: msg.owner,
+        },
+    )?;
+
+    msg.curve_config.validate()?;
+
+    RESTRICTION.save(deps.storage, &msg.restriction)?;
+    CURVE_CONFIG.save(deps.storage, &msg.curve_config)?;
+
+    Ok(resp)
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let ctx = ExecuteContext::new(deps, info, env);
+    match msg {
+        ExecuteMsg::AMPReceive(pkt) => {
+            ADOContract::default().execute_amp_receive(ctx, pkt, handle_execute)
+        }
+        _ => handle_execute(ctx, msg),
+    }
+}
+
+fn handle_execute(mut ctx: ExecuteContext, msg: ExecuteMsg) -> Result<Response, ContractError> {
+    let action = msg.as_ref().to_string();
+
+    let action_response = call_action(
+        &mut ctx.deps,
+        &ctx.info,
+        &ctx.env,
+        &ctx.amp_ctx,
+        msg.as_ref(),
+    )?;
+
+    let res = match msg.clone() {
+        ExecuteMsg::UpdateCurveConfig { curve_config } => {
+            execute_update_curve_config(ctx, curve_config, action)
+        }
+        ExecuteMsg::UpdateRestriction { restriction } => {
+            execute_update_restriction(ctx, restriction, action)
+        }
+        ExecuteMsg::Reset {} => execute_reset(ctx, action),
+        _ => ADOContract::default().execute(ctx, msg),
+    }?;
+
+    Ok(res
+        .add_submessages(action_response.messages)
+        .add_attributes(action_response.attributes)
+        .add_events(action_response.events))
+}
+
+pub fn execute_update_curve_config(
+    ctx: ExecuteContext,
+    curve_config: CurveConfig,
+    action: String,
```

Comment:
Is this just for the response's attribute? If so, just hardcode the action's name in the attribute

Reply:
I resolved it